changelog/issue-7128.md

```diff
@@ -0,0 +1,6 @@
+audience: users
+level: patch
+reference: issue 7128
+---
+
+Use `/usr/sbin/deluser --remove-home` instead of `/usr/sbin/deluser --remove-all-files` when deleting a user on Linux. This ensures that caches that may still be owned (in whole or in part) by the task user are not deleted.
```

Comment:
```suggestion
Generic Worker multiuser engine on Linux now uses `/usr/sbin/deluser --remove-home` instead of `/usr/sbin/deluser --remove-all-files` when deleting previous task users. This ensures that caches that may still be owned (in whole or in part) by the task user are not deleted.
```

Sorry I missed this in the first review. I think it wasn't clear this is a Generic Worker fix otherwise, and that it related to task users being deleted on workers.